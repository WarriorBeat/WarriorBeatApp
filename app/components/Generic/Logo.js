/**
 * Logo.js
 * WarriorBeat Logo Image
 * componenents/Generic
 */

import React from "react"
import Image from "react-native-fast-image"
import brandMedia from "config/assets"
import { ViewPropTypes, StyleSheet } from "react-native"
import { Shape } from "components/styles"
import { PropTypes } from "mobx-react/native"

const Logo = (props) => {
  const { style, resize, size } = props
  const styles = StyleSheet.flatten([Shape.circle(size), style])
  return <Image source={brandMedia.warrior_head} style={styles} resizeMode={resize} />
}

Logo.propTypes = {
  style: ViewPropTypes.style,
  resize: Image.propTypes.resizeMode,
  size: PropTypes.string,
}

Logo.defaultProps = {
  style: {},
  resize: Image.resizeMode.contain,
  size: "30%",
}

export default Logo
