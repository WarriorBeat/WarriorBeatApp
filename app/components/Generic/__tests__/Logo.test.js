/**
 * __tests__/Logo.test.js
 * Component Test
 * tests
 */
/* eslint-disable */

import React from "react"
import { shallow } from "react-native-testing-library"
import Logo from "../Logo"

test("should render correctly", () => {
  const tree = shallow(<Logo />)
  expect(tree).toMatchSnapshot()
})
