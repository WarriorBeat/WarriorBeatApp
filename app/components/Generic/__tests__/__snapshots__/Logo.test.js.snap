// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
Object {
  "output": <ForwardRef
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../app/assets/brand/main.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 112.5,
        "height": 225,
        "width": 225,
      }
    }
  />,
}
`;
